### PR TITLE
feat: add SSO

### PR DESCRIPTION
**Pull request by oleobal**
## Description

Add mechanism to handle a new authentication flow: single sign-on (SSO) through OpenID Connect (OIDC).

Changes:
 - on the login page, we query the `/info` endpoint to see whether SSO is available, and if so present a button. All the button does is redirect to the backend at `/oidc/authenticate?next={some frontend page}`. The backend does everything else
 - add the current username in the header menu, because with SSO the username is not obvious
 - add an "API tokens" modal (in the header menu) so SSO users can use the library (they don't have a password).
   
   In the future we will deprecate logging in on the command line, and that modal will be a nice interface for creating and managing API tokens. Probably.

Issues: _(to be fixed in a future PR, presumably)_
 - active tokens make too many requests and should go through Redux
 - API tokens modal is ugly

## How to test

Only the actual login requires special set-up.

Set up an SSO-capable backend: https://github.com/Substra/substra-backend/pull/609

Then just edit `vite.config.ts` to point to `http://substra-backend.org-1.com` (without the port number, because we're hosting on the cluster rather than in isolated mode). That should be it.

It's also possible to test the frontend in isolated mode but that requires more set-up on the backend so I'll let you figure it out.

## To do

- [x] update changelog
- [x] ~~investigate login/logout redirects (using the logout button and then logging in again works weird)~~ can't reproduce

### CODE REVIEW

File: src/modules/bearerTokens/BearerTokenApi.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+const API_ACTIVE_TOKENS_URL = `/active-api-tokens`;
+
+const parseToken = (object: RawBearerTokenT): BearerTokenT => {
+    return {
+        created_at: new Date(object.created_at),
+        expires_at: new Date(object.expires_at),
+    };
+};
+
+const parseNewToken = (object: RawNewBearerTokenT): NewBearerTokenT => {
```
Comment (Milouu) [inline src/modules/bearerTokens/BearerTokenApi.ts:20]: Same, should be in a utils file

File: src/modules/bearerTokens/BearerTokenApi.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+export const requestToken = () => {
+    return API.authenticatedGet(API_TOKEN_URL).then((response) => {
+        return parseNewToken(response.data);
+    });
+};
+
+export const listActiveTokens = () => {
+    return API.authenticatedGet(API_ACTIVE_TOKENS_URL).then((response) => {
```
Comment (Milouu) [inline src/modules/bearerTokens/BearerTokenApi.ts:35]: Same, then part is usually not handled in API file

File: src/routes/login/components/LoginForm.tsx [hunk 3]
```diff
@@ -75,6 +75,29 @@ const LoginForm = (): JSX.Element => {
                 <Text fontWeight="semibold" fontSize="3xl" marginBottom="4">
                     Login to {organizationId}
                 </Text>
+
+                {authInfo?.oidc && (
+                    <div>
+                        <a
```
Comment (Milouu) [inline src/routes/login/components/LoginForm.tsx:81]: Shoud use wouter `<Link>` for internal links

**Review by Milouu — CHANGES_REQUESTED** [review]

File: src/components/ApiToken.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                <VStack align="left">
+                    <Text fontSize="sm">
+                        {'Created ' + token.created_at.toLocaleString()}
+                    </Text>
+                    <Text
+                        fontSize="sm"
+                        color={
+                            token.expires_at < new Date()
+                                ? 'red.500'
+                                : undefined
```
[the remark this answers is not in the document]
Comment (oleobal) [reply src/components/ApiToken.tsx:57]: I want to inherit whatever the theme sets for normal text color.

File: src/routes/login/components/LoginForm.tsx (hunk 3 above)
Comment (oleobal) [reply src/routes/login/components/LoginForm.tsx]: This is an external link to the backend, but I can use chakra's

**Review by Milouu — APPROVED** [review]
Nice work :) Could only test the Google auth

File: src/components/ApiToken.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                <Center width="120px">
+                    <Tooltip
+                        hasArrow
+                        label="This will disable your existing token"
+                        bg="red.500"
+                        isDisabled={tokenIsExpired || 'token' in apiToken}
+                    >
+                        <Button
+                            onClick={getNewToken}
+                            disabled={'token' in apiToken}
```
Comment (jmorel) [inline src/components/ApiToken.tsx:90]: You should use isDisabled here.

Also, I'm not clear on why token.apiToken being defined should disable the button.

**Review by jmorel — DISMISSED** [review]

**Pull request merged**